tracer/src/Datadog.Trace/SpanContextExtractor.cs

```diff
@@ -38,7 +38,7 @@ public SpanContextExtractor()
             TelemetryFactory.Metrics.Record(PublicApiUsage.SpanContextExtractor_Extract);
             var spanContext = SpanContextPropagator.Instance.Extract(carrier, getter);
             if (spanContext is not null
-             && Tracer.Instance.TracerManager.DataStreamsManager is { IsEnabled: true } dsm
+             && Tracer.InternalInstance.TracerManager.DataStreamsManager is { IsEnabled: true } dsm
```

Comment:
Yep this is called from customer code, but the implementation requires getting the automatic tracer instance in order to use the right propagation settings. In 6cbe1cc527c49bdd0c26fcb9bdf56afc1508086c I noted this dependency, and I propose we proxy the `SpanContextExtractor` and `SpanContextInjector` as part of our larger changes, so that it will be safe to use `Tracer.InternalInstance`

Reply:
I... _think_ I understand 😅 So we'll have a `SpanContextExtractorInternal` and `SpanContextInjectorInternal` too subsequently? 🤔

Reply:
Something like that! 😅 I admit this is a little hand-wavy, but because the `SpanContextExtractor` and `SpanContextInjector` need to do behind-the-scenes work like DSM and different propagation formats, let's assume that we are already inside the automatic assembly.

In this PR I'm not planning to actually do any rewriting yet, but I've left the TODO comments so we know that we need to give the user a `SpanContextExtractor`/`SpanContextInjector` object from the automatic assembly, which is the same thing we're planning to do with the `Tracer` object